spec/unit/filecache_spec.rb

```diff
@@ -84,6 +84,24 @@ def write_file(file, contents)
         end
       end
 
+      it "sets the encoding to UTF-8 when reading a file" do
+        Dir.mktmpdir do |dir|
+          file = File.join(dir, "testing")
+          write_file(file, "my data")
+          File.expects(:read).with(file, :encoding => 'BOM|UTF-8')
+          @cache.read_file(file)
+        end
+      end
+
+      it "reads a file with unicode characters" do
+        Dir.mktmpdir do |dir|
+          file = File.join(dir, "testing")
+          write_file(file, "☃")
```

Comment:
I typically use the "\u" style notation or take an actually byte array and convert to utf8 - some examples:

https://github.com/puppetlabs/puppet/blob/79c593921a51e9d59ba7644900dc127ebfe0d61e/spec/unit/pops/loaders/dependency_loader_spec.rb#L63
https://github.com/puppetlabs/puppet/blob/75d183f97ff60d45e15291ce8319d827cdb7d555/spec/unit/util_spec.rb#L123

That helps to prevent any weird parsing errors when Ruby reads files..

If you look through the Puppet code, we've sprinkled a lot of `rune` characters around - they're 3 byte UTF-8 characters (your Unicode snowman is also a 3-byte character, which is good - http://www.fileformat.info/info/unicode/char/2603/index.htm)